data-generation/src/main/scala/thesis/Landy.scala

```diff
@@ -36,35 +37,74 @@ class Landy(graph: LandyAttributeGraph) extends TemporalGraph[LandyVertexPayload
    * @param interval Inclusive interval
    * @return Tuple with the activated entities
    */
-  override def activatedEntities(interval: Interval): (RDD[VertexId], RDD[EdgeId]) = throw new NotImplementedError()
+  override def activatedEntities(interval: Interval): (RDD[VertexId], RDD[EdgeId]) = {
+    val firstLocalVertices = localVertices // Remove global vertices
+      .map(vertex => (vertex._2.id, vertex._2)) // Get the global vertex ID
+      .groupByKey() // Group all
+      .map(vertex => (vertex._1, getEarliest(vertex._2.toSeq))) // Get first local vertex
+
+    val vertices = firstLocalVertices.filter(vertex =>
+      vertex._2.validFrom >= interval.start &&
```

Comment:
Vi gjør en del "er dette timestampet inni det intervallet?"-type funksjoner. Kanskje det skulle vært refakturer til en egen funksjon.

```
     isInInclusiveInterval(interval:Interval)(timestamp:Instant):Boolean
```

```suggestion
      .filter(isInInclusiveInterval(interval, _._2.validFrom))
```

(Suggestion-greia blir litt feil)

Reply:
Ja, enig!